coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -54,38 +66,41 @@ class ShardKeyRegexPlanner(dataset: Dataset,
   private def generateExec(logicalPlan: LogicalPlan, nonMetricShardKeyFilters: Seq[ColumnFilter],
                            qContext: QueryContext): Seq[ExecPlan] = {
     val queryParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    shardKeyMatcherFn(ShardKeyMatcher(nonMetricShardKeyFilters, queryParams.promQl)).map { result =>
-        val newQueryParams = queryParams.copy(promQl = result.query)
+    shardKeyMatcher(nonMetricShardKeyFilters).map { result =>
+        val newLogicalPlan = logicalPlan.replaceFilters(result)
+        val newQueryParams = queryParams.copy(promQl = LogicalPlanUtils.logicalPlanToQuery(newLogicalPlan))
         val newQueryContext = qContext.copy(origQueryParams = newQueryParams)
-        queryPlanner.materialize(logicalPlan.replaceFilters(result.columnFilters), newQueryContext)
+        queryPlanner.materialize(logicalPlan.replaceFilters(result), newQueryContext)
       }
   }
 
-  private def materializeBinaryJoin(binaryJoin: BinaryJoin, qContext: QueryContext): ExecPlan = {
+  private def materializeBinaryJoin(binaryJoin: BinaryJoin, qContext: QueryContext): PlanResult = {
    if (getNonMetricShardKeyFilters(binaryJoin).forall(_.forall(f => !f.filter.isInstanceOf[EqualsRegex] &&
-     !f.filter.isInstanceOf[NotEqualsRegex]))) queryPlanner.materialize(binaryJoin, qContext)
+     !f.filter.isInstanceOf[NotEqualsRegex]))) PlanResult(Seq(queryPlanner.materialize(binaryJoin, qContext)))
    else throw new UnsupportedOperationException("Regex not supported for Binary Join")
   }
 
-  private def materializeAggregate(aggregate: Aggregate, queryContext: QueryContext): ExecPlan = {
+  private def materializeAggregate(aggregate: Aggregate, queryContext: QueryContext): PlanResult = {
```

Comment:
An example of PromQL query that each of the materialize* methods addresses as comment would be very useful for the reader.

Why was special handling needed in this class, as opposed to pushing them up to your new trait ? Please answer these question here.